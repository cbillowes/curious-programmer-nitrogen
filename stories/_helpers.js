import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page