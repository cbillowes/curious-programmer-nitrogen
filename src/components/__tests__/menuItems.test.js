import React from 'react'
import MenuItems from '../menuItems'
import { getSnapshot } from './_helpers'

const domain = `https://curiousprogrammer.dev`
const toggleOnClick = function (e) { }

describe(`Menu items`, () => {
  it(`should not set any page as active when the page does not exist`, () => {
    jsdom.reconfigure({
      url: `${domain}/blogging`,
    })
    const component = (
      <MenuItems
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should set the blog page to active for /blog`, () => {
    jsdom.reconfigure({
      url: `${domain}/blog`,
    })
    const component = (
      <MenuItems
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should set the blog page to active for /blog/`, () => {
    jsdom.reconfigure({
      url: `${domain}/blog/`,
    })
    const component = (
      <MenuItems
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should set the blog page to active for a given post`, () => {
    jsdom.reconfigure({
      url: `${domain}/blog/dont-panic`,
    })
    const component = (
      <MenuItems
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should set another page to active in the navigation items`, () => {
    jsdom.reconfigure({
      url: `${domain}/privacy-policy/`,
    })
    const component = (
      <MenuItems
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })
})
