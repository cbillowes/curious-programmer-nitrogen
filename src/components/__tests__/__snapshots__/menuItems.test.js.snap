// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu items should not set any page as active when the page does not exist 1`] = `
<div
  className="menu"
>
   
  <a
    className="menu-item"
    href="/blog"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Blog
  </a>
   
   
  <a
    className="menu-item"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
   
   
  <a
    className="menu-item"
    href="/archives"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Archives
  </a>
   
   
  <a
    className="menu-item"
    href="/tags"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tags
  </a>
   
   
  <a
    className="menu-item"
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Credits & Resources
  </a>
   
   
  <a
    className="menu-item"
    href="/privacy-policy"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Privacy Policy
  </a>
   
   
  <a
    className="menu-item"
    href="/license"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    License
  </a>
   
   
  <a
    className="menu-item"
    href="https://github.com/cbillowes"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Github
  </a>
   
   
  <a
    className="menu-item"
    href="https://www.linkedin.com/in/cbouwer/"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    LinkedIn
  </a>
   
</div>
`;

exports[`Menu items should set another page to active in the navigation items 1`] = `
<div
  className="menu"
>
   
  <a
    className="menu-item"
    href="/blog"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Blog
  </a>
   
   
  <a
    className="menu-item"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
   
   
  <a
    className="menu-item"
    href="/archives"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Archives
  </a>
   
   
  <a
    className="menu-item"
    href="/tags"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tags
  </a>
   
   
  <a
    className="menu-item"
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Credits & Resources
  </a>
   
  <a
    className="menu-item active"
    href="/#"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex="0"
  >
    Privacy Policy
  </a>
   
  <a
    className="menu-item"
    href="/license"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    License
  </a>
   
   
  <a
    className="menu-item"
    href="https://github.com/cbillowes"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Github
  </a>
   
   
  <a
    className="menu-item"
    href="https://www.linkedin.com/in/cbouwer/"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    LinkedIn
  </a>
   
</div>
`;

exports[`Menu items should set the blog page to active for /blog 1`] = `
<div
  className="menu"
>
  <a
    className="menu-item active"
    href="/#"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex="0"
  >
    Blog
  </a>
   
  <a
    className="menu-item"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
   
   
  <a
    className="menu-item"
    href="/archives"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Archives
  </a>
   
   
  <a
    className="menu-item"
    href="/tags"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tags
  </a>
   
   
  <a
    className="menu-item"
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Credits & Resources
  </a>
   
   
  <a
    className="menu-item"
    href="/privacy-policy"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Privacy Policy
  </a>
   
   
  <a
    className="menu-item"
    href="/license"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    License
  </a>
   
   
  <a
    className="menu-item"
    href="https://github.com/cbillowes"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Github
  </a>
   
   
  <a
    className="menu-item"
    href="https://www.linkedin.com/in/cbouwer/"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    LinkedIn
  </a>
   
</div>
`;

exports[`Menu items should set the blog page to active for /blog/ 1`] = `
<div
  className="menu"
>
  <a
    className="menu-item active"
    href="/#"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex="0"
  >
    Blog
  </a>
   
  <a
    className="menu-item"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
   
   
  <a
    className="menu-item"
    href="/archives"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Archives
  </a>
   
   
  <a
    className="menu-item"
    href="/tags"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tags
  </a>
   
   
  <a
    className="menu-item"
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Credits & Resources
  </a>
   
   
  <a
    className="menu-item"
    href="/privacy-policy"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Privacy Policy
  </a>
   
   
  <a
    className="menu-item"
    href="/license"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    License
  </a>
   
   
  <a
    className="menu-item"
    href="https://github.com/cbillowes"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Github
  </a>
   
   
  <a
    className="menu-item"
    href="https://www.linkedin.com/in/cbouwer/"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    LinkedIn
  </a>
   
</div>
`;

exports[`Menu items should set the blog page to active for a given post 1`] = `
<div
  className="menu"
>
  <a
    className="menu-item active"
    href="/#"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex="0"
  >
    Blog
  </a>
   
  <a
    className="menu-item"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
   
   
  <a
    className="menu-item"
    href="/archives"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Archives
  </a>
   
   
  <a
    className="menu-item"
    href="/tags"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tags
  </a>
   
   
  <a
    className="menu-item"
    href="/credits"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Credits & Resources
  </a>
   
   
  <a
    className="menu-item"
    href="/privacy-policy"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Privacy Policy
  </a>
   
   
  <a
    className="menu-item"
    href="/license"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    License
  </a>
   
   
  <a
    className="menu-item"
    href="https://github.com/cbillowes"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Github
  </a>
   
   
  <a
    className="menu-item"
    href="https://www.linkedin.com/in/cbouwer/"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    LinkedIn
  </a>
   
</div>
`;
